### PR TITLE
C/C++ highlights: Small improvements for type definitions

### PR DESCRIPTION
**Pull request by theHamsta**
Definition of types by:
 - using (C++)
 - plain struct

Reference of types

### CODE REVIEW

File: queries/c/locals.scm [hunk 1]
```diff
@@ -26,9 +26,12 @@
   declarator: (field_identifier) @definition.field)
 (type_definition
   declarator: (type_identifier) @definition.type)
+(struct_specifier
+  name: (type_identifier) @definition.type)
 
 ;; References
 (identifier) @reference
+(type_identifier) @reference
```
Comment (vigoux) [inline queries/c/locals.scm:34]: I don't think that type_identifiers are actually references.

Comment (theHamsta) [reply queries/c/locals.scm:34]: What is a reference for you? It references a type identifier when we can have type definitions we should also have type references. Or maybe add reference.type?

Comment (theHamsta) [reply queries/c/locals.scm:34]: I'll have a look again at which positions that node type can occur

Comment (theHamsta) [reply queries/c/locals.scm:34]: `type_reference` is the `Foo` in `Foo a = create_foo();` So references to type names.

**Pull request merged**